sdk/keyvault/platform-matrix.json

```diff
@@ -0,0 +1,17 @@
+{
+  "displayNames": {
+    "@{ enableHsm = $true }": "HSM"
+  },
+  "include": [
+    {
+      "Agent": {
+        "ubuntu-18.04": {
+          "OSVmImage": "MMSUbuntu18.04",
+          "Pool": "azsdk-pool-mms-ubuntu-1804-general"
+        }
+      },
+      "ArmTemplateParameters": "@{ enableHsm = $true }",
+      "TestTargetFramework": [ "netcoreapp2.1" ]
```

Comment:
I am not familiar with `perform-matrix` files but when looking around the Java repo I found a couple more that had this. Maybe @danieljurek or @mitchdenny can tell us if this is correct.

Reply:
@benbp created it for .NET so we only run 1 set of tests against Managed HSM (per-hour cost), which is a much more expensive resource than Key Vault (per-transaction cost). The preview constraints for instances also was a problem.

Reply:
Still, @alzimmermsft is right that you shouldn't use a .NET target framework moniker (TFM) for the Java repo.

Reply:
Gotcha, will remove that and specify a Java version and HTTP client. Thanks to both for the feedback.

Reply:
There are a few other required matrix variables you'll need for the java tests to work (pulled from the [base matrix](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/stages/platform-matrix.json)). I think something like this would be closer to working:

```
{
  "displayNames": {
    "@{ enableHsm = $true }": "HSM"
  },
  "matrix": {
    "Agent": {
      "ubuntu-18.04": {
        "OSVmImage": "MMSUbuntu18.04",
        "Pool": "azsdk-pool-mms-ubuntu-1804-general"
      }
    },
    "JavaTestVersion": "1.11",
    "AZURE_TEST_HTTP_CLIENTS": "netty",
    "TestGoals": "surefire:test",
    "TestOptions": "",
    "ArmTemplateParameters": "@{ enableHsm = $true }"
  }
}
```